<launch>
    <arg name="WristPitchPWM_CH"    value="/PWM/CH0" />

    <node name="PWM_HAT" pkg="mavric" type="PCA9685_PWM_HAT.py" respawn="true">
      <param name="container" value="PWM/" />
      <param name="control_mode" value="DutyCycle" />
      <param name="frequency" value="250" />
      
      <!-- addresses start at 0x40 (64) and can be configured on the board -->
      <param name="address" value="64" />
      
      <!-- we want to run as fast as possible here -->
      <!-- <param name="clk_error" value="1.0893246187363834422657952069717" /> -->
      <param name="clk_error" value="1.04166667" />
     </node>
      
    <node name="WristPitch_Scale" pkg="mavric" type="L298N_Controller.py">
      <param name="input"  value="WristPitch" />
      <param name="output" value="$(arg WristPitchPWM_CH)" />
      <param name="slope"      value="0.01" />
      <param name="intercept"  value="0" />
      <param name="forward"  value="35" />
      <param name="backward"  value="37" />
    </node>
    
    <node name="PIDController" pkg="pid" type="controller" respawn="true" >
      <param name="Kp" value="1.0"/>
      <param name="Ki" value="0.75"/>
      <param name="Kd" value="0.1"/>
      <param name="upper_limit" value="100"/>
      <param name="lower_limit" value="-100"/>
      <param name="windup_limit" value="10"/>
      <param name="max_loop_frequency" value="100.0" />
      <param name="min_loop_frequency" value="100.0" />
      <remap from="setpoint" to="/DesiredPos" />
      <remap from="control_effort" to="/WristPitch" />
      <remap from="state" to="/Steer_Feedback" />
      <remap from="pid_enable" to="/ArmEnable" />
    </node>

    <node name="Encoder" pkg="mavric" type="mavric_Encoder_Feedback" respawn="true"/>
</launch>
